File: src/main/java/us/fatehi/whatacharacter/part1/EscapeSequences.java
/*

What a Character
https://github.com/sualeh/What-a-Character
Copyright (c) 2016-2024, Sualeh Fatehi.

This library is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

This work is licensed under the Creative Commons Attribution-ShareAlike
4.0 International License. To view a copy of this license, visit
http://creativecommons.org/licenses/by-sa/4.0/.

*/
package us.fatehi.whatacharacter.part1;


public class EscapeSequences
{

  private static void printUnicode(char ch) {
    System.out.println(toUnicode(ch));
  }

  private static String toUnicode(char ch) {
    return String.format("\\u%04x", (int) ch);
  }

  public static void main(String[] args)
  {

    // Character escape sequences

    printUnicode('\t');
    printUnicode('\b');
    printUnicode('\n');
    printUnicode('\r');
    printUnicode('\f');
    printUnicode('\'');
    printUnicode('\"');
    printUnicode('\\');
  }

}

File: src/main/python/part1/escape_sequences.py
"""

What a Character
https://github.com/sualeh/What-a-Character
Copyright (c) 2016-2024, Sualeh Fatehi.

This library is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

This work is licensed under the Creative Commons Attribution-ShareAlike
4.0 International License. To view a copy of this license, visit
http://creativecommons.org/licenses/by-sa/4.0/.

"""


def print_unicode(ch):
    print(to_unicode(ch))


def to_unicode(ch):
    return "\\u{:04x}".format(ord(ch))


def main():
    
    # Character escape sequences

    print_unicode("\t")
    print_unicode("\b")
    print_unicode("\n")
    print_unicode("\r")
    print_unicode("\f")
    print_unicode("'")
    print_unicode('"')
    print_unicode("\\")


if __name__ == "__main__":
    main()
